modular_zubbers/code/game/objects/structures/lewd_portals.dm

```diff
@@ -0,0 +1,447 @@
+			switch(dir)
+				if(NORTH)
+					current_mob.pixel_y += wallstuck_offset_amount
+				if(SOUTH)
+					current_mob.pixel_y += -wallstuck_offset_amount
+					current_mob.transform = turn(current_mob.transform, ROTATION_FLIP)
+				if(EAST)
+					current_mob.pixel_x += wallstuck_offset_amount
+					current_mob.transform = turn(current_mob.transform, ROTATION_COUNTERCLOCKWISE)
+				if(WEST)
+					current_mob.pixel_x += -wallstuck_offset_amount
+					current_mob.transform = turn(current_mob.transform, ROTATION_CLOCKWISE)
+	else
+		unbuckle_all_mobs()
+	..()
+
+///Algorithm that calculates what a mob head offset should be based on mob scale.
+/obj/structure/lewd_portal/proc/offset_algorithm()
+	var/transform_scale_height = mob_scale_manager.scale_y
+	if(transform_scale_height <= 1)
+		wallstuck_offset_amount = (-30 * transform_scale_height) + 42
+	else
+		wallstuck_offset_amount = (-24 * transform_scale_height) + 36
+	wallstuck_offset_amount = clamp(round(wallstuck_offset_amount), 0, 18)
+
+
+///Hides the buckled mob's penis
+/obj/structure/lewd_portal/proc/hide_penis()
+	SIGNAL_HANDLER
+	var/obj/item/organ/genital/penis/affected_penis = current_mob.get_organ_slot(ORGAN_SLOT_PENIS) //Stolen from Strapon code, this is bad we should probably have a cleaner way
+	affected_penis?.visibility_preference = GENITAL_NEVER_SHOW
+	current_mob.update_body()
+	affected_penis?.visibility_preference = initial_genital_visibility //These seems weird but I need to hide the penis when people use the gloryhole while maintaining its visability so it can be interacted with.
+
+///Removes everything besides the head for the buckled mob, used in wallstuck mode
+/obj/structure/lewd_portal/proc/head_only()
+	SIGNAL_HANDLER
+	current_mob.cut_overlays()
+	current_mob.update_body_parts_head_only()
+	var/obj/item/clothing/glasses = current_mob.glasses
+	if(glasses)
+		current_mob.update_worn_glasses()
+	current_mob.remove_overlay(BODY_ADJ_LAYER)
+	current_mob.remove_overlay(BODY_LAYER)
+	current_mob.remove_overlay(HANDS_LAYER)
+	var/obj/item/bodypart/head/mob_head = current_mob.get_bodypart(BODY_ZONE_HEAD)
+	if(mob_head.head_flags & HEAD_EYESPRITES)
+		var/obj/item/organ/eyes/eye_organ = current_mob.get_organ_slot(ORGAN_SLOT_EYES)
+		if(eye_organ)
+			eye_organ.refresh(call_update = FALSE)
+			current_mob.overlays_standing[BODY_LAYER] += eye_organ.generate_body_overlay(current_mob)
+			current_mob.apply_overlay(BODY_LAYER)
+
+/obj/structure/lewd_portal/post_unbuckle_mob(mob/living/unbuckled_mob)
+	UnregisterSignal(current_mob, list(COMSIG_MOB_POST_EQUIP, COMSIG_HUMAN_UNEQUIPPED_ITEM, COMSIG_HUMAN_TOGGLE_UNDERWEAR, COMSIG_MOB_HANDCUFFED, COMSIG_MOB_EMOTE, COMSIG_EMOTE_OVERLAY_EXPIRE, COMSIG_HUMAN_ADJUST_AROUSAL, COMSIG_HUMAN_TOGGLE_GENITALS))
+	visible_message("[current_mob] exits the [src]")
+	current_mob = null
+	mob_scale_manager = null
+	qdel(relayed_body)
```

Comment:
```suggestion
	QDEL_NULL(relayed_body)
```
QDEL_NULL also sets the var to null after calling qdel, we want this because otherwise the reference to it stays in `relayed_body` and it can never fully clean up